runtime/interpreter/interpreter.go

```diff
@@ -3551,6 +3624,36 @@ var runtimeTypeConstructors = []runtimeTypeConstructor{
 			},
 		),
 	},
+	{
+		name: "InclusiveRangeType",
+		converter: NewUnmeteredHostFunctionValue(
+			sema.InclusiveRangeTypeFunctionType,
+			func(invocation Invocation) Value {
+				typeValue, ok := invocation.Arguments[0].(TypeValue)
+				if !ok {
+					panic(errors.NewUnreachableError())
+				}
+
+				ty := typeValue.Type
+				// InclusiveRanges must hold integers
+				elemSemaTy := invocation.Interpreter.MustConvertStaticToSemaType(ty)
+				if !elemSemaTy.Tag().BelongsTo(sema.IntegerTypeTag) {
```

Comment:
I didn't intend this to be an optimization. I did this because I thought this is the only way to do the check. I am doing something similar in `value_range.go` to detect if an integer is `unsigned` based on the suggestion from Supun here: https://github.com/onflow/cadence/pull/2523#discussion_r1256520701

Do you recommend using `sema.IsSubType` instead?

Reply:
We might be able to use `IsSameTypeKind` (probably not `IsSubType`, because `Void` is a subtype of every type).

In the `value_range.go` use-case, it checks if a type is an unsigned integer, but there is no equivalent sema-type (i.e: no `UnsignedIntegerType`), so using `IsSameTypeKind`/`IsSubType` in that case wasn't really possible. That was kind of a special case where the tag had to be used, or rather, using the tag was convenient.

However, here since we are checking whether the type is a subtype of `Integer` type, and given there is a `sema.IntegerType`, maybe it would be better to use `IsSameTypeKind` for consistency (all other places use IsSameTypeKind/IsSubType for sub-type checking).

Reply:
Thanks Supun and Bastian. Updated to use `IsSameTypeKind` in 7904b9544.